src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java

```diff
@@ -1645,7 +1678,36 @@ public File getImportRoot() {
     }
 
     public File getUploadRoot() {
+        String value = null;
         try {
+            value = System.getProperty(UPLOAD_ROOT_KEY);
+            if (value == null) {
+                value = System.getenv(UPLOAD_ROOT_KEY);
+            }
+        } catch (Throwable ex) {
+            if (LOGGER.isLoggable(Level.FINEST)) {
+                LOGGER.log(Level.FINEST, "Could not access system property '" + UPLOAD_ROOT_KEY + "': " + ex);
+            }
+        }
+        
+        if (configFile != null && configFile.isModified()) {
+            try {
+                props = configFile.read();
+            } catch (Exception e) {
+                LOGGER.log(Level.FINEST,
+                        "Could not find any '" + PROPERTYFILENAME + "' property file.", e);
+            }
+        }
+        
+        if (props != null && value == null) {
+            value = props.getProperty(UPLOAD_ROOT_KEY);
+        }
+        
+        try {
+            if(value != null) {
+                Resource uploadsRoot = Resources.fromPath(value);
+                return Resources.directory(uploadsRoot, !Resources.exists(uploadsRoot));
```

Comment:
What if someone wants to use an absolute path to keep data outside of the data directory? (quite common)
Suggestion: use a fixed location relative to the data directory by default, and for configured locations, check first if they are absolute paths.

Reply:
This check is done already by "Resources.fromPath" method.

On my tests it works either with relative and absolute paths.